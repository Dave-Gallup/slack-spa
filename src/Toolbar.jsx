// @flow
import { Dropdown, Image, Menu } from 'semantic-ui-react';
import React from 'react';
import slack from './images/slack_icon.png';
import './index.css';

function importAll(r) {
  return r.keys().reduce(
    (map, path) => {
      map[path.replace('./', '')] = r(path);
      return map;
    },
    {}
  );
}

const images = importAll(
  (require: any).context('./images/emojis', false, /\.(png|jpg)$/)
);

const emojis = Object.keys(images).reduce(
  (map, path) => {
    map[path.replace(/\.(png|jpg)$/, '')] = path;
    return map;
  },
  {}
);

const scoreToEmoji = {
  '1': 'angry',
  '2': 'sad',
  '3': 'disappointed',
  '4': 'neutral',
  '5': 'smile',
  '6': 'happy',
};

type Props = {
  color: string,
  scoreStatus: boolean,
  score: mixed,
};

export default function Toolbar({ color, score, scoreStatus }: Props) {
  const menuClasses = `ui ${color} inverted menu`;

  const currentEmoji =
    !score ? scoreToEmoji[4] : scoreToEmoji[parseInt(score, 10)];
  return (
    <Menu size="small" className={menuClasses}>
      <Menu.Item className="ui button">
        <Image avatar src={slack} />
      </Menu.Item>
      <Dropdown item text="Channel name">
        <Dropdown.Menu>
          <Dropdown.Item>Team 1</Dropdown.Item>
          <Dropdown.Item>Team 2</Dropdown.Item>
          <Dropdown.Item>Team 3</Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
      <Menu.Menu position="right">
        <Menu.Item className="ui button">
          <Image
            avatar
            src={images[emojis[currentEmoji]]}
          />
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}
