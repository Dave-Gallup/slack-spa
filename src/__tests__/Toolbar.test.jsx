import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Toolbar from '../Toolbar';

describe('Renders toolbar', () => {
  it('should render a toolbar', () => {
    const toolbar = shallow(<Toolbar />);
    expect(toJson(toolbar)).toMatchSnapshot();
  });
});
