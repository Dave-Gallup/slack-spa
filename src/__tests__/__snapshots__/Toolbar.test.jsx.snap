// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders toolbar should render a toolbar 1`] = `
<Menu
  className="ui teal inverted menu"
  size="small"
>
  <MenuItem
    className="ui button"
  >
    <Icon
      as="i"
      className="slack icon large"
      name="slack"
    />
  </MenuItem>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    icon="dropdown"
    item={true}
    minCharacters={1}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    searchInput="text"
    selectOnBlur={true}
    text="Channel name"
  >
    <DropdownMenu>
      <DropdownItem>
        Team 1
      </DropdownItem>
      <DropdownItem>
        Team 2
      </DropdownItem>
      <DropdownItem>
        Team 3
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
  <MenuMenu
    position="right"
  >
    <MenuItem
      className="ui button"
    >
      <Icon
        as="i"
        className="smile icon large"
        name="smile"
      />
    </MenuItem>
  </MenuMenu>
</Menu>
`;
